src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,133 @@
+
+        if (sections.isEmpty()) {
+            sections.add(newSection);
+            return this;
+        }
+
+        validateNewSection(newSection);
+
+        Section switchSection = getSwitchTarget(newSection);
+        switchSectionValue(switchSection, newSection);
+        sections.add(newSection);
+        sortSections();
+
+        return this;
+    }
+
+    private void switchSectionValue(Section previousSection, Section newSection) {
+        if (previousSection != null) {
+            previousSection.switchValue(newSection);
+        }
+    }
+
+    private Section getSwitchTarget(Section newSection) {
+
+        Section previousSection = findSectionWithUpStation(newSection);
+
+        if (previousSection == null) {
+            previousSection = findSectionWithDownStation(newSection);
+        }
+
+        return previousSection;
+    }
+
+
+    private Section findSectionWithUpStation(Section newSection) {
+        return sections.stream()
+                .filter(section -> section.hasUpStation(newSection.getUpStation()))
+                .findFirst()
+                .orElse(null);
+    }
+
+    private Section findSectionWithDownStation(Section newSection) {
+        return sections.stream()
+                .filter(section -> section.hasDownStation(newSection.getDownStation()))
+                .findFirst()
+                .orElse(null);
+    }
+
+    public List<Section> getSectionList() {
+        return sections;
+    }
+
+    public void validateNewSection(Section newSection) {
+
+        boolean containsUpStation = sections.stream()
+                .anyMatch(section -> section.containsStation(newSection.getUpStation()));
+
+        boolean containsDownStation = sections.stream()
+                .anyMatch(section -> section.containsStation(newSection.getDownStation()));
+
+        if (containsUpStation && containsDownStation) {
+            System.out.println(STATIONS_ALREADY_EXIT_EXCEPTION);
+            throw new IllegalArgumentException(STATIONS_ALREADY_EXIT_EXCEPTION);
+        }
+
+        if (!containsUpStation && !containsDownStation) {
+            System.out.println(STATIONS_DO_NOT_EXIST_EXCEPTION);
+            throw new IllegalArgumentException(STATIONS_DO_NOT_EXIST_EXCEPTION);
+        }
+    }
+
+    private void sortSections() {
+        //상행종점역 찾기
+        Section firstSection = sections.stream()
+                .filter(section -> getPreviousSection(section) == null)
+                .findAny()
+                .get();
+
+        //상행종점역부터 정렬
+        List<Section> sortedSections = new ArrayList<>();
+        sortedSections.add(firstSection);
+
+        Section previousSection = firstSection;
+
+        while (sortedSections.size() != sections.size()) {
+            Section nextSection = getNextSection(previousSection);
+            sortedSections.add(nextSection);
+            previousSection = nextSection;
+
+        }
+
+        sections = sortedSections;
```

Comment:
```suggestion
        return sortedSections;
```
`sections`를 변경하지 않고 `sortedSections`를 리턴하는건 어떨까요? 😀

Reply:
이 부분은 정렬 메서드에서는 `List<Section>`인  `sortedSections` 를 반환하도록 수정하였지만, 제가 `Line에` `sections` 를 리턴하도록 설계했기 때문에 어디에선가는 해당 값을 바꿔야 하므로 마지막 `getSortedSections` 메서드에서는 치환해주었습니다!
혹시 내부적으로 sections 주소값의 치환을 지양하라는 의미이신건지 고민이 되네요. 🤔